cmd/kubeadm/app/cmd/init.go

```diff
@@ -0,0 +1,199 @@
+		`(optional) Specify range of IP addresses for the pod network. If set, the control plane will automatically allocate CIDRs for every node.`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.Services.DNSDomain, "service-dns-domain", kubeadmapi.DefaultServiceDNSDomain,
+		`(optional) Use alternative domain for services, e.g. "myorg.internal"`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.CloudProvider, "cloud-provider", "",
+		`(optional) Enable a specific cloud provider features (external load-balancers, storage, etc), e.g. "gce"`,
+	)
+	cmd.PersistentFlags().BoolVar(
+		&s.InitFlags.Schedulable, "schedule-pods-here", false,
+		`(optional) Allow to schedule workload to the node`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.Versions.Kubernetes, "use-kubernetes-version", kubeadmapi.DefaultKubernetesVersion,
+		`(optional) Choose a specific Kubernetes version for the control plane`,
+	)
+
+	// TODO (phase1+) @errordeveloper make the flags below not show up in --help but rather on --advanced-help
+	cmd.PersistentFlags().StringSliceVar(
+		&s.InitFlags.API.Etcd.ExternalEndpoints, "external-etcd-endpoints", []string{},
+		`(optional) etcd endpoints to use, in case you have an external cluster.`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.API.Etcd.ExternalCAFile, "external-etcd-cafile", "",
+		`(optional) etcd certificate authority certificate file."`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.API.Etcd.ExternalCertFile, "external-etcd-certfile", "",
+		`(optional) etcd client certificate file."`,
+	)
+	cmd.PersistentFlags().StringVar(
+		&s.InitFlags.API.Etcd.ExternalKeyFile, "external-etcd-keyfile", "",
+		`(optional) etcd client key file."`,
+	)
+
+	return cmd
+}
+
+// RunInit executes master node provisioning, including certificates, needed static pod manifests, etc.
+func RunInit(out io.Writer, cmd *cobra.Command, args []string, s *kubeadmapi.KubeadmConfig, advertiseAddrs *[]string) error {
+	// Auto-detect the IP
+	if len(*advertiseAddrs) == 0 {
+		// TODO(phase1+) perhaps we could actually grab eth0 and eth1
+		ip, err := netutil.ChooseHostInterface()
+		if err != nil {
+			return err
+		}
+		s.InitFlags.API.AdvertiseAddrs = []net.IP{ip}
+	} else {
+		for _, i := range *advertiseAddrs {
+			addr := net.ParseIP(i)
+			if addr == nil {
+				return fmt.Errorf("<cmd/init> failed to parse flag (%q) as an IP address", "--api-advertise-addresses="+i)
+			}
+			s.InitFlags.API.AdvertiseAddrs = append(s.InitFlags.API.AdvertiseAddrs, addr)
+		}
+	}
+
+	if s.InitFlags.CloudProvider != "" {
+		// TODO(phase2) we should be able to auto-detect it and check whether things like IAM roles are correct
+		if _, ok := kubeadmapi.SupportedCloudProviders[s.InitFlags.CloudProvider]; !ok {
+			return fmt.Errorf("<cmd/init> cloud provider %q is not supported, you can use any of %v, or leave it unset", s.InitFlags.CloudProvider, kubeadmapi.ListOfCloudProviders)
```

Comment:
Why are we adding `<cmd/init>` to these?  why aren't we adding `<cmd/init>` to the errors below?